src/hooks/useQuoteManager.js

```diff
@@ -0,0 +1,207 @@
+        method: 'get',
+        url: `${process.env.REACT_APP_API_URL}/property?searchText=${cleanAddress}&state=${state}`,
+      });
+      setSearchResults(response.data.result.IndexResult);
+    } catch (error) {
+      throw error;
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  async function createQuote(
+    { companyCode, state, product, propertyId },
+    setView
+  ) {
+    setLoading(true);
+    try {
+      const response = await axios({
+        headers: { Authorization: AUTH_TOKEN },
+        method: 'post',
+        url: `${process.env.REACT_APP_API_URL}/quote`,
+        data: {
+          companyCode,
+          state,
+          product,
+          propertyId,
+        },
+      });
+      setQuoteResult(response.data.result);
+      setSearchResults(undefined);
+      setView('viewQuote');
+    } catch (error) {
+      setError(error);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  async function updateQuote({ quote, input }) {
+    setLoading(true);
+    try {
+      formatForSubmit(input);
+
+      const response = await axios({
+        headers: { Authorization: AUTH_TOKEN },
+        method: 'put',
+        url: `${process.env.REACT_APP_API_URL}/quote/${quote.quoteNumber}`,
+        data: {
+          quote,
+          input,
+        },
+      });
+      setQuoteResult(response.data.result);
+    } catch (error) {
+      setError(error);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  async function retrieveQuote(quoteNumber, setView) {
+    setLoading(true);
+    try {
+      const response = await axios({
+        headers: { Authorization: AUTH_TOKEN },
+        method: 'get',
+        url: `${process.env.REACT_APP_API_URL}/quote/${quoteNumber}`,
+      });
+      setQuoteResult(response.data.result);
+      setView('viewQuote');
+    } catch (error) {
+      setError(error);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  async function sendApplication(quoteNumber) {
+    setLoading(true);
+    try {
+      await axios({
+        headers: { Authorization: AUTH_TOKEN },
+        method: 'post',
+        url: `${process.env.REACT_APP_API_URL}/sendApplication`,
+        data: {
+          quoteNumber,
+          sendType: 'docusign',
+        },
+      });
+      setApplicationSuccess(true);
+    } catch (error) {
+      setError(error);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  function resetQuoteState() {
+    setError(undefined);
+    setSearchResults(undefined);
+    setQuoteResult(undefined);
+    setApplicationSuccess(undefined);
+  }
+
+  function resetSearchResults() {
+    setSearchResults(undefined);
+  }
+
+  return {
+    createQuote,
+    updateQuote,
+    fetchQuote: retrieveQuote,
+    searchAddress,
+    sendApplication,
+    resetQuoteState,
+    resetSearchResults,
+    loading,
+    error,
+    searchResults,
+    quoteResult,
+    applicationSuccess,
+  };
+}
+
+function formatForSubmit(doc) {
```

Comment:
Please add // TODO here to call out that we would ultimately like anything in this function to be fixed via the backend.

Reply:
Added comment